integration-tests/search.test.ts

```diff
@@ -513,3 +514,242 @@ describe('search', () => {
         });
     });
 });
+
+describe('searches with patient permission levels', () => {
+    let adminClient: AxiosInstance;
+    let encounterResource: any;
+
+    beforeAll(async () => {
+        adminClient = await getFhirClient('fhirUser user/*.*', true);
+        const encounterResourceJson = {
+            resourceType: 'Encounter',
+            status: 'in-progress',
+            class: {
+                system: 'http://terminology.hl7.org/CodeSystem/v3-ActCode',
+                code: 'IMP',
+                display: 'inpatient encounter',
+            },
+            subject: {
+                reference: `Patient/92e0d921-bb19-4cae-a3cc-9d3c5bcf7a39`,
+                display: 'Sherlock Holmes',
+            },
+        };
+        encounterResource = (await adminClient.post('Encounter', encounterResourceJson)).data;
+        await waitForResourceToBeSearchable(adminClient, encounterResource);
+    });
+
+    afterAll(async () => {
+        await adminClient.delete(`Encounter/${encounterResource.id}`);
+    });
+
+    test('tests with only patient.read scope', async () => {
+        const client = await getFhirClient('fhirUser patient/Patient.read launch/patient profile openid', false);
+        const sherlockHolmes = (await adminClient.get(`Patient/${idsOfFhirResources.sherlockHolmes}`)).data;
+
+        const getPatient = await client.get('Patient');
+        expect(getPatient.data.total).toBe(2); // Should return Mycroft Holmes as a seealso
+        expect(getPatient.data.entry[1].resource).toMatchObject(sherlockHolmes);
+
+        await expectResourceToNotBePartOfSearchResults(
+            client,
+            { url: 'Patient', params: { _include: 'Patient:organization' } },
+            {
+                resourceType: 'Organization',
+            },
+        );
+
+        await expectResourceToNotBePartOfSearchResults(
+            client,
+            { url: 'Patient', params: { _revinclude: 'Encounter:subject' } },
+            encounterResource,
+        );
+
+        // failing searches (no scopes for Observation, so should reject)
+        await expect(client.get('Observation')).rejects.toMatchObject({
+            response: { status: 401 },
+        });
+
+        await expect(client.get('Observation', { params: { _include: 'Observation:subject' } })).rejects.toMatchObject({
+            response: { status: 401 },
+        });
+    });
+
+    test('tests with additional patient scopes', async () => {
+        const orgClient = await getFhirClient(
+            'fhirUser patient/Patient.read patient/Organization.read launch/patient profile openid',
+            false,
+        );
+        const encClient = await getFhirClient(
+            'fhirUser patient/Patient.read patient/Encounter.read launch/patient profile openid',
+            false,
+        );
+
+        await expectResourceToNotBePartOfSearchResults(
```

Comment:
Add a quick comment as to why this should not be in search result -- mainly that the patient has scope access to the Organization but not resource access